packages/stryker/test/unit/reporters/ScoreReporterSpec.ts

```diff
@@ -0,0 +1,137 @@
+      // Arrange
+      setupEnvironmentVariables();
+      sut = new ScoreReporter(new Config, badgeClientMock as any);
+
+      // Act
+      sut.onScoreCalculated(scoreResult({ mutationScore: 79.10 }));
+      await sut.wrapUp();
+
+      // Assert
+      const report: StrykerBadgeReport = {
+        api_key: '12345',
+        repository_slug: 'github/stryker-mutator/stryker',
+        branch: 'master',
+        mutation_score: 79.10,
+        report_data: []
+      };
+
+      expect(badgeClientMock.postStrykerBadgeReport).to.have.been.calledWith(report);
+      expect(log.warn).to.have.not.been.called;
+    });
+  });
+
+  describe('when the build is on pull request', () => {
+    it('log that pull requests are not reported to the badge report server', async () => {
+      // Arrange
+      setupEnvironmentVariables({ pullRequest: '1' });
+      sut = new ScoreReporter(badgeClientMock as any);
+
+      // Act
+      sut.onScoreCalculated(scoreResult({ mutationScore: 79.10 }));
+      await sut.wrapUp();
+
+      // Assert
+      expect(badgeClientMock.postStrykerBadgeReport).to.have.not.been.called;
+      expect(log.warn).to.have.been.calledWithMatch('Pull requests will not be reported to the badge report server');
+      expect(log.warn).to.have.been.calledWithMatch('Mutation score is not reported to the badge report server');
+    });
+  });
+
+  describe('when the repository is unknown', () => {
+    it('log that the environment variable TRAVIS_REPO_SLUG should be set', async () => {
+      // Arrange
+      setupEnvironmentVariables({ repository: undefined });
+      sut = new ScoreReporter(badgeClientMock as any);
+
+      // Act
+      sut.onScoreCalculated(scoreResult({ mutationScore: 79.10 }));
+      await sut.wrapUp();
+
+      // Assert
+      expect(badgeClientMock.postStrykerBadgeReport).to.have.not.been.called;
+      expect(log.warn).to.have.been.calledWithMatch('Missing environment variable TRAVIS_REPO_SLUG to determine the repository');
+      expect(log.warn).to.have.been.calledWithMatch('Mutation score is not reported to the badge report server');
+    });
+  });
+
+  describe('when the branch is unknown', () => {
+    it('log that the environment variable TRAVIS_BRANCH should be set', async () => {
+      // Arrange
+      setupEnvironmentVariables({ branch: undefined });
+      sut = new ScoreReporter(badgeClientMock as any);
+
+      // Act
+      sut.onScoreCalculated(scoreResult({
+        mutationScore: 79.10
+      }));
+      await sut.wrapUp();
+
+      // Assert
+      expect(badgeClientMock.postStrykerBadgeReport).to.have.not.been.called;
+      expect(log.warn).to.have.been.calledWithMatch('Missing environment variable TRAVIS_BRANCH to determine the branch');
+      expect(log.warn).to.have.been.calledWithMatch('Mutation score is not reported to the badge report server');
+    });
+  });
+
+  describe('When the Stryker badge API key is unknown', () => {
```

Comment:
`it('should log a info/warning if the api key is missing')`

Reply:
Done